test/unit/deadline_client/config/test_config_file.py

```diff
@@ -273,32 +273,84 @@ def test_default_log_level():
     platform.system() != "Windows",
     reason="This test is for testing file permission changes in Windows.",
 )
-def test_windows_config_file_permissions(fresh_deadline_config) -> None:
-    config_file_path = config_file.get_config_file_path()
-    parent_dir = config_file_path.parent
-    subprocess.run(
-        [
-            "icacls",
-            str(parent_dir),
-            "/grant",
-            "Everyone:(OI)(CI)(F)",
-            "/T",
-        ],
-        check=True,
-    )
+def test_reset_directory_permissions_windows() -> None:
+    """
+    Asserts the _reset_directory_permissions_windows configures the provided
+    folder with access only to the active user, the domain admin, and SYSTEM.
+    """
+    # GIVEN
+    import ntsecuritycon
 
-    config_file.set_setting("defaults.aws_profile_name", "goodguyprofile")
+    win32security_mock = MagicMock()
+    sys.modules["win32security"] = win32security_mock
```

Comment:
Nit: I don't think this functionality needs mocking-based tests at all. It could test within a temporary directory to create a few directories with the wrong permissions, and validate that the function resets them successfully. This would decouple the tests from the implementation details and let those change more easily later.

Reply:
Changed the test to remove the file system mocks and instead validate that the permissions are what we expect.